packages/svelte2tsx/src/svelte2tsx/nodes/ExportedNames.ts

```diff
@@ -184,7 +184,10 @@ export class ExportedNames {
                     generic_arg.end + this.astOffset,
                     node.parent.pos + this.astOffset
                 );
-                this.str.appendRight(generic_arg.end + this.astOffset, this.$props.type);
+                this.str.appendRight(
+                    generic_arg.end + this.astOffset,
+                    surroundWithIgnoreComments(this.$props.type)
```

Comment:
```suggestion
                    // so that semantic tokens ignore it, preventing an overlap of tokens
                    surroundWithIgnoreComments(this.$props.type)
```